docs/roslyn_best_practices.md

```diff
@@ -0,0 +1,181 @@
+# Roslyn Best Practices
+
+We use [Roslyn Analyzers](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview) highlight areas of code that will need to be refactored. By default our goal is to pair these Analyzers with Code Fixers to automate as much of the upgrade workflow as possible.
```

Comment:
It may be good to call out some definitions and describe what analyzers and codefixers are at a high level to level set the concept with potential new contributors (and send them off to the docs site for a more detailed description).

Maybe include terms like: Analyzer, Code Fixer, Language agnostic, etc